cmd/kiya/main.go

```diff
@@ -169,7 +176,18 @@ func getBackend(ctx context.Context, p *backend.Profile) (backend.Backend, error
 		}
 
 		return backend.NewGSM(gsmClient), nil
-
+	case "akv":
+		cred, err := azidentity.NewDefaultAzureCredential(nil)
+		if err != nil {
+			log.Fatal(err)
+		}
+		client, err := azsecrets.NewClient(p.VaultUrl, cred, nil)
+		if err != nil {
+			log.Fatal(err)
+		}
+		return backend.NewAKV(client), nil
+	case "file":
+		return backend.NewFileStore(p.Location, p.ProjectID), nil
```

Comment:
I'm more in favor of passing the password to the New or the struct instead of a set-method like above. This also would not change the interface implemented by all. Feels more of an implementation detail of the file store backend. 

@Geurtje @emicklei wdyt?

Reply:
the ALL commands pollution , I would prefer to avoid.  one way or another this PW fetch code need to be somewhere.

Reply:
think there is some option inbetween, where there is an empty password variable before the select case, where it's filled in for "get" and "put". The password variable (either filled or not filled) will be passed to the struct or new constructor. @jroosing wdyt?